backend/api/Services/LocalizationService.cs

```diff
@@ -0,0 +1,247 @@
+                logger.LogError("{Message}", errorMessage);
+                throw new InstallationNotFoundException(errorMessage);
+            }
+
+            if (robotInstallation != missionInstallation)
+            {
+                string errorMessage = $"The robot {robot.Name} is on installation {robotInstallation.Name} which is not the same as the mission installation {missionInstallation.Name}";
+                logger.LogError("{Message}", errorMessage);
+                throw new MissionException(errorMessage);
+            }
+        }
+
+        public async Task<string?> EnsureRobotIsCorrectlyLocalized(Robot robot, MissionRun missionRun)
+        {
+            if (missionRun.Area is null)
+            {
+                string errorMessage = $"There was no area associated with mission run {missionRun.Id}";
+                logger.LogError("{Message}", errorMessage);
+                throw new AreaNotFoundException(errorMessage);
+            }
+
+            string? localizationMissionRunId = null;
+
+            if (!await RobotIsLocalized(robot.Id)) { localizationMissionRunId = await StartLocalizationMissionInArea(robot.Id, missionRun.Area.Id); }
+
+            if (!await RobotIsOnSameDeckAsMission(robot.Id, missionRun.Area.Id))
+            {
+                string errorMessage = $"The new mission run {missionRun.Id} will not be started as the robot is not localized on the same deck as the mission";
+                logger.LogError("{Message}", errorMessage);
+                throw new RobotLocalizationException(errorMessage);
+            }
+
+            logger.LogWarning("{Message}", $"Localization mission run ID is {localizationMissionRunId}");
+
+            return localizationMissionRunId;
+        }
+
+        public async Task<bool> RobotIsLocalized(string robotId)
+        {
+            var robot = await robotService.ReadById(robotId);
+            if (robot is null)
+            {
+                string errorMessage = $"Robot with ID: {robotId} was not found in the database";
+                logger.LogError("{Message}", errorMessage);
+                throw new RobotNotFoundException(errorMessage);
+            }
+
+            return robot.CurrentArea is not null;
+        }
+
+        public async Task EnsureRobotWasCorrectlyLocalizedInPreviousMissionRun(string robotId)
+        {
+            var robot = await robotService.ReadById(robotId);
+            if (robot == null)
+            {
+                string errorMessage = $"Robot with ID: {robotId} was not found in the database";
+                logger.LogError("{Message}", errorMessage);
+                throw new RobotNotFoundException(errorMessage);
+            }
+
+            if (await missionRunService.OngoingMission(robot.Id)) { await WaitForLocalizationMissionStatusToBeUpdated(robot); }
```

Comment:
Catch all relevant exceptions

Reply:
The intended behaviour is that these are thrown by this function and caught by the caller of `WaitForLocalizationMissionStatusToBeUpdated`